File: OrchidCore/src/main/java/com/eden/orchid/impl/themes/components/ReadmeComponent.java
package com.eden.orchid.impl.themes.components;

import com.eden.orchid.api.OrchidContext;
import com.eden.orchid.api.resources.resource.OrchidResource;
import com.eden.orchid.api.theme.components.OrchidComponent;

import javax.inject.Inject;

public class ReadmeComponent extends OrchidComponent {

    @Inject
    public ReadmeComponent(OrchidContext context) {
        super(50, "readme", context);
    }

    public String getContent() {
        OrchidResource readmeResource = context.findClosestFile("readme");
        if (readmeResource != null) {
            return context.compile(readmeResource.getReference().getExtension(), readmeResource.getContent(), this);
        }
        return null;
    }
}

File: OrchidCore/src/main/kotlin/com/eden/orchid/impl/themes/components/ReadmeComponent.kt
package com.eden.orchid.impl.themes.components

import com.eden.orchid.api.options.annotations.Description
import com.eden.orchid.api.theme.components.OrchidComponent

@Description(value = "Locate and display your project's Readme file.", name = "Readme")
class ReadmeComponent : OrchidComponent("readme") {
    val content: String? get() = context.findClosestFile("readme")?.compileContent(context, this)
}
